common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -4631,7 +4646,7 @@ static void NativeCrypto_ASN1_TIME_to_Calendar(JNIEnv* env, jclass, jlong asn1Ti
 
     bssl::UniquePtr<ASN1_GENERALIZEDTIME> gen(ASN1_TIME_to_generalizedtime(asn1Time, nullptr));
     if (gen.get() == nullptr) {
-        conscrypt::jniutil::throwNullPointerException(env, "asn1Time == null");
+        conscrypt::jniutil::throwNullPointerException(env, "gen == null");
```

Comment:
This fixes a typo, but making this an NPE seems a little error-prone. Rather than split into `ASN1_TIME_check` and `ASN1_TIME_to_Calendar`, why not make this line throw some kind of parsing exception and mark `ASN1_TIME_to_Calendar` as throwing in NativeCrypto.java? That means the compiler will catch whether `ASN1_TIME_to_Calendar` is only called when it is safe to.

Reply:
Yeah, that sounds better.  Updated.